proof/refine/ARM/IpcCancel_R.thy

```diff
@@ -3956,31 +4212,32 @@ lemma cancelBadgedSends_corres:
                   apply (rule corres_guard_imp)
                     apply (subst bind_assoc[symmetric])
                     apply (rule corres_guard_imp)
-                      apply (fold restartThreadIfNoFault_def[simplified K_bind_def isNothing_def])
+                      apply (fold restartThreadIfNoFault_def[simplified])
                       apply (rule corres_split_deprecated[OF _ restart_thread_if_no_fault_corres])
                         apply (rule corres_return_eq_same, simp)
                        apply (rule hoare_TrueI[where P=\<top>])
                       apply (rule hoare_TrueI[where P=\<top>])
                      apply simp+
                 apply (wpsimp wp: gts_wp)
                apply (wpsimp wp: gts_wp')
-              apply (fastforce simp: st_tcb_def2 st_tcb_at_refs_of_rev
-                              dest!: state_refs_of_elemD )
+              apply (clarsimp simp: st_tcb_def2 st_tcb_at_refs_of_rev valid_sched_def
+                              dest!: state_refs_of_elemD)
              apply (clarsimp simp: st_tcb_def2 st_tcb_at_refs_of_rev)
-
```

Comment:
This isn't directly to do with this PR, but it could be nice to make a style decision about whitespace in proofs. It can be hard to tell sometimes whether it's an accident that can be removed or if someone put it in to add structure to a complicated proof. @lsf37 @Xaphiosis @mktnk3 @michaelmcinerney thoughts?

Reply:
One possibility is that when whitespace is being used for structure then there could be a comment as well.